<a href="https://colab.research.google.com/github/aqmalio/Data-Science-2026/blob/main/pertemuan2_aqmal_250401020149.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

* Nama: Aqmal
* Nim: 250401020149
* Kelas: IF405



In [ ]:
import pandas as pd
import numpy as np

In [ ]:
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/refs/heads/master/titanic.csv"
df = pd.read_csv(URL)

In [ ]:
print(df.shape, df.columns.tolist())

(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


In [ ]:
print(df.isnull().sum())

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


In [ ]:
wk1 = df[(df['Sex'] == 'female') & (df['Pclass']==1)]
print(wk1['Survived'].mean(), wk1['Age'].mean())

0.9680851063829787 34.61176470588235


In [ ]:
hasil = df.groupby('Pclass').agg(
    total=('PassengerId', 'count'),
    selamat=('Survived', 'mean'),
).round(3)
print(hasil)

        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242


In [ ]:
print(df.groupby('Pclass')['Survived'].mean())

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64


### Kesimpulan

**Apa yang Dipelajari:**
- Cara melakukan eksplorasi dasar pada dataset Titanic menggunakan pandas.
- Cara mengidentifikasi data yang hilang dan melakukan agregasi data berdasarkan kategori.

**Temuan Utama:**
- **Korelasi Kelas & Keselamatan:** Penumpang Kelas 1 memiliki tingkat kelangsungan hidup tertinggi (63%), sedangkan Kelas 3 paling rendah (24%).
- **Segmen Spesifik:** Penumpang perempuan di Kelas 1 memiliki peluang selamat yang sangat tinggi (96.8%).

**Keterbatasan & Pertanyaan:**
- Terdapat banyak data kosong pada kolom `Age` dan `Cabin` yang membatasi analisis.
- Muncul pertanyaan: Apakah faktor harga tiket (`Fare`) juga berkorelasi langsung dengan keselamatan dalam kelas yang sama?